zipkin/src/main/java/zipkin/internal/v2/codec/Encoder.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2015-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.internal.v2.codec;
+
+import zipkin.Span;
+import zipkin.internal.JsonCodec;
+import zipkin.internal.Span2;
+
+import static zipkin.internal.v2.codec.Span2JsonAdapters.SPAN_WRITER;
+
+/**
+ * @param <S> type of the span, usually {@link Span}
+ */
+public interface Encoder<S> {
```

Comment:
This, Encoding and MessageEncoder were taken from zipkin-reporter-java, which afterwards can be a place to host just the implementations